### ISSUE TITLE
/crypto/dso/dso_dlfcn.c:118:filename(libz.so): libz.so: cannot open shared object file: No such file or directory

### ISSUE THREAD

**Issue by tpgxyz** [openssl/openssl#19539]
Hi,
looks like openssl3 is linking to development zlib library name/soname. Basically in RPM world a *.so (i.e. /usr/lib64/libz.so) files goes to development subpackage while *.so.%{major} (i.e. /usr/lib64/libz.so.1) are distributed with main library/program to reduce depencencies bloat.

Please notice that **openss speed** dies with error of not finding **libz.so** while it should look fro **libz.so.1**
```
[tpg@omv-rockpro64 ~]# openssl speed -evp aes-128-cbc                                                                                                                                                            
Doing AES-128-CBC for 3s on 16 size blocks: 67162828 AES-128-CBC's in 3.00s
Doing AES-128-CBC for 3s on 64 size blocks: 41881115 AES-128-CBC's in 3.00s
Doing AES-128-CBC for 3s on 256 size blocks: 16098478 AES-128-CBC's in 3.00s
Doing AES-128-CBC for 3s on 1024 size blocks: 4545001 AES-128-CBC's in 3.00s
Doing AES-128-CBC for 3s on 8192 size blocks: 603758 AES-128-CBC's in 3.00s
Doing AES-128-CBC for 3s on 16384 size blocks: 302434 AES-128-CBC's in 3.00s
version: 3.0.6
built on: Thu Oct 27 16:05:05 2022 UTC
options: bn(64,64)
compiler: clang -fPIC -pthread -Wa,--noexecstack -Qunused-arguments -Os -fomit-frame-pointer -g3 -gdwarf-4 -Wstrict-aliasing=2 -pipe -Wformat -Werror=format-security -D_FORTIFY_SOURCE=2 -fstack-protector-all --param=ssp-buffer-size=4 -flto -O3 -fno-strict-aliasing -fprofile-use=/builddir/build/BUILD/openssl-3.0.6/build/openssl-llvm.profdata -mllvm -polly -mllvm -polly-run-dce -mllvm -polly-run-inliner -mllvm -polly-isl-arg=--no-schedule-serialize-sccs -mllvm -polly-ast-use-context -mllvm -polly-detect-keep-going -mllvm -polly-vectorizer=stripmine -DOPENSSL_USE_NODELETE -DOPENSSL_PIC -DOPENSSL_BUILDING_OPENSSL -DZLIB -DZLIB_SHARED -DNDEBUG
CPUINFO: OPENSSL_armcap=0xbd
The 'numbers' are in 1000s of bytes per second processed.
type             16 bytes     64 bytes    256 bytes   1024 bytes   8192 bytes  16384 bytes
AES-128-CBC     358201.75k   893463.79k  1373736.79k  1551360.34k  1648661.85k  1651692.89k
2040469FFFFF0000:error:12800067:DSO support routines:dlfcn_load:could not load the shared library:../crypto/dso/dso_dlfcn.c:118:filename(libz.so): libz.so: cannot open shared object file: No such file or directory
2040469FFFFF0000:error:12800067:DSO support routines:DSO_load:could not load the shared library:../crypto/dso/dso_lib.c:152:

```
Other important information:

```
[tpg@omv-rockpro64 ~]# rpm -qf /usr/lib64/libz.so.1                                                                                                                                                              
lib64z1-2.0.6-8.aarch64
```

```
[tpg@omv-rockpro64 ~]# rpm -qf $(command -v openssl)                                                                                                                                                             
openssl-3.0.6-2.aarch64
```
```
[tpg@omv-rockpro64 ~]# cat /etc/os-release                                                                                                                                                                       
NAME="OpenMandriva Lx"
VERSION="4.90 (Nickel) Cooker"
ID="openmandriva"
VERSION_ID="4.90"
PRETTY_NAME="OpenMandriva Lx 4.90 (Nickel) Cooker"
BUILD_ID="20221008.17"
VERSION_CODENAME="nickel"
ANSI_COLOR="1;43"
LOGO="openmandriva"
CPE_NAME="cpe:/o:openmandriva:openmandriva_lx:4.90"
HOME_URL="http://openmandriva.org/"
BUG_REPORT_URL="https://github.com/OpenMandrivaAssociation/distribution/issues/"
SUPPORT_URL="https://forum.openmandriva.org"
PRIVACY_POLICY_URL="https://www.openmandriva.org/tos"
```

OpenSSL  RPM sources https://github.com/OpenMandrivaAssociation/openssl/blob/master/openssl.spec

Build with Configure ith zlib instead of zlib-dynamic fixes the issue

**Comment by t8m** [comment]
Then do not use the zlib-dynamic build option. That option makes the zlib support to use libz as a "plugin".

Unfortunately there is no easy way how to know what is the soname version to use. Perhaps we could add a way how to override it via a defined macro. But that would be a new feature.

**Comment by t8m** [comment]
So, to make it clear, if you want libcrypto to properly link zlib as a shared library, use just the enable-zlib.

**Comment by tpgxyz** [comment]
Hi @t8m yes i figured this out to pass `zlib` instead of `zlib-dynamic`. This means in general practice for RPM distributed software where software is split to development bits, is to use `zlib` to overcome any possible missing libraries and errors.

**Comment by nhorman** [comment]
marking as inactive, will be closed at the end of the 3.4 dev cycle barring further updates
